[Package]: yazi

### Why is it worth to add this package?

Yazi is a blazing fast terminal file manager written in Rust, based on async I/O.

### Home page URL

https://yazi-rs.github.io/

### Source code URL

https://github.com/sxyazi/yazi

### Packaging policy acknowledgement

- [X] I certify that I have read [Termux Packaging Policy](https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#packaging-policy) and understand that my request will be denied in case of violation.

### Additional information

_No response_

I have tried to compile the program but got the following errors.

```
error[E0433]: failed to resolve: use of undeclared crate or module `platform`
--> /home/builder/.cargo/registry/src/index.crates.io-6f17d22bba15001f/trash-3.2.1/src/lib.rs:67:35
|
67 |         Self { platform_specific: platform::PlatformTrashContext::new() }
|                                   ^^^^^^^^ use of undeclared crate or module `platform`

error[E0599]: no method named `delete_all_canonicalized` found for reference `&TrashContext` in the current scope
--> /home/builder/.cargo/registry/src/index.crates.io-6f17d22bba15001f/trash-3.2.1/src/lib.rs:112:14
|
112 |         self.delete_all_canonicalized(full_paths)
|              ^^^^^^^^^^^^^^^^^^^^^^^^ method not found in `&TrashContext`

error[E0433]: failed to resolve: use of undeclared crate or module `platform`
--> /home/builder/.cargo/registry/src/index.crates.io-6f17d22bba15001f/trash-3.2.1/src/lib.rs:63:24
|
63 |     platform_specific: platform::PlatformTrashContext,
|                        ^^^^^^^^ use of undeclared crate or module `platform`

Some errors have detailed explanations: E0433, E0599.
```

--- a/packages/yazi/build.sh
+++ b/packages/yazi/build.sh
@@ -1,0 +1,30 @@
+TERMUX_PKG_HOMEPAGE=https://yazi-rs.github.io/
+TERMUX_PKG_DESCRIPTION="Blazing fast terminal file manager written in Rust, based on async I/O"
+TERMUX_PKG_LICENSE="MIT"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION="0.2.2"
+TERMUX_PKG_SRCURL=https://github.com/sxyazi/yazi/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz
+TERMUX_PKG_SHA256=ce830fc312fc7a9515abefbbc71c8d1a46515257e9d1c56165cf6ff2fa5404c7
+TERMUX_PKG_AUTO_UPDATE=true
+TERMUX_PKG_BUILD_IN_SRC=true
+
+termux_step_make() {
+	termux_setup_rust
+	YAZI_GEN_COMPLETIONS=true cargo build --jobs $TERMUX_MAKE_PROCESSES --target $CARGO_TARGET_NAME --release
+}
+
+termux_step_make_install() {
+	install -Dm700 -t $TERMUX_PREFIX/bin target/${CARGO_TARGET_NAME}/release/yazi
+
+	cd yazi-config/completions
+	install -Dm644 "${TERMUX_PKG_NAME}".bash "${TERMUX_PREFIX}"/share/bash-completion/completions/"${TERMUX_PKG_NAME}".bash
+	install -Dm644 _"${TERMUX_PKG_NAME}" "${TERMUX_PREFIX}"/share/zsh/site-functions/_"${TERMUX_PKG_NAME}"
+	install -Dm644 "${TERMUX_PKG_NAME}".fish "${TERMUX_PREFIX}"/share/fish/vendor_completions.d/"${TERMUX_PKG_NAME}".fish
+}
+
+termux_step_create_debscripts() {
+	cat <<- POSTINST_EOF > ./postinst
+	#!$TERMUX_PREFIX/bin/sh
+	echo "Please change font from termux-styling addon"
+	POSTINST_EOF
+}

--- a/packages/yazi/pr-600.patch
+++ b/packages/yazi/pr-600.patch
@@ -1,0 +1,38 @@
+From d17c8910a60adc0133e38f727c40f3e676fea5dd Mon Sep 17 00:00:00 2001
+From: sxyazi <sxyazi@gmail.com>
+Date: Tue, 30 Jan 2024 01:37:15 +0800
+Subject: [PATCH] feat: make `trash` crate optional on Android
+
+---
+ yazi-scheduler/Cargo.toml       | 4 +++-
+ yazi-scheduler/src/file/file.rs | 2 +-
+ 2 files changed, 4 insertions(+), 2 deletions(-)
+
+diff --git a/yazi-scheduler/Cargo.toml b/yazi-scheduler/Cargo.toml
+index be0a828f..94aecb94 100644
+--- a/yazi-scheduler/Cargo.toml
++++ b/yazi-scheduler/Cargo.toml
+@@ -24,7 +24,9 @@ parking_lot            = "^0"
+ regex                  = "^1"
+ tokio                  = { version = "^1", features = [ "parking_lot", "rt-multi-thread" ] }
+ tokio-stream           = "^0"
+-trash                  = "^3"
+ 
+ # Logging
+ tracing = { version = "^0", features = [ "max_level_debug", "release_max_level_warn" ] }
++
++[target.'cfg(not(target_os = "android"))'.dependencies]
++trash = "^3"
+diff --git a/yazi-scheduler/src/file/file.rs b/yazi-scheduler/src/file/file.rs
+index dc869440..dec74cf8 100644
+--- a/yazi-scheduler/src/file/file.rs
++++ b/yazi-scheduler/src/file/file.rs
+@@ -122,7 +122,7 @@ impl File {
+ 					ctx.set_delete_method(DeleteMethod::NsFileManager);
+ 					ctx.delete(&task.target)?;
+ 				}
+-				#[cfg(not(target_os = "macos"))]
++				#[cfg(all(not(target_os = "macos"), not(target_os = "android")))]
+ 				{
+ 					trash::delete(&task.target)?;
+ 				}

